### PR TITLE
feat: add filtering to expenses

### PR DESCRIPTION
**Pull request by Edu58**

### CODE REVIEW

**Comment by Edu58** [comment]
@mohitkhatri06  It is ready for review. You can test and give feedback

**Comment by mohitkhatri06** [comment]
after applying the filter add pagination, as for now it is taking long time to response after providing the range. And also if we can see the range what we had chosen (as for now after clicking the filter button we are not able to see which range I chosen), there should be some different button or something that can clear the filter. Let me know if you have any queries.

**Comment by bharathkumarchowdary** [comment]
This filtering doesn't look good


image

**Comment by bharathkumarchowdary** [comment]
> after applying the filter add pagination, as for now it is taking long time to response after providing the range. And also if we can see the range what we had chosen (as for now after clicking the filter button we are not able to see which range I chosen), there should be some different button or something that can clear the filter. Let me know if you have any queries.

I can't understand your comment @mohitkhatri06 , Can you please explain in layman terms?

**Comment by mohitkhatri06** [comment]
@bharathkumarchowdary as for now after we choose start and end date and click on the filter button, all records comes in a single page, whole list at once seems like.
Apart from that after selecting the range and click on the button it doesn't show that which range I have chosen or for which filter its showing the results.

**Review by ivangarcilazo — COMMENTED** [review]
Yes, I think the same with Mohit and Bharath, just a improve in a design, I think like to make the form to row direction instead of column. And when I apply the filter, the dates are deleted, It's better if the dates selected dont are deleted, for better user experience. 👍

**Comment by Edu58** [comment]
@mohitkhatri06  Check the design now. For the pagination. I have to make changes to the API so it can serve paginated data also

**Comment by mohitkhatri06** [comment]
@Edu58 still when we enter the range and click on the filter button the selected range disappears.

**Comment by Edu58** [comment]
@mohitkhatri06 
@bharathkumarchowdary 
Fixed the clearing of the date range on filtering. Also added a clear button to clear the filters once done.

fixedexpenses

File: src/components/Expenses/Expenses.jsx [hunk 1]
```diff
@@ -134,38 +171,81 @@ const Expenses = ({ Context }) => {
...
+            labelDisplayedRows={() => ''}
+          />
+          <IconButton
+            onClick={() => {
+              setPage((prevPage) => Math.min(prevPage + 3, 6));
+            }}
+          >
+            <SkipNext />
+          </IconButton>
+          {page * 10 + 1} - {page * 10 + 10} of 70
```
Comment (mohitkhatri06) [inline src/components/Expenses/Expenses.jsx:205]: instead of that 70 constant we should have the data which we get from the end point

**Comment by mohitkhatri06** [comment]
> @mohitkhatri06 Check the design now. For the pagination. I have to make changes to the API so it can serve paginated data also expensesfix

as we need to do some changes in pagination (backend) , approving the following PR,

**Review by mohitkhatri06 — DISMISSED** [review]

**Review by bharathkumarchowdary — APPROVED** [review]

**Review by ivangarcilazo — DISMISSED** [review]
Everything looks good and functional.

**Review by mohitkhatri06 — APPROVED** [review]

**Review by ivangarcilazo — APPROVED** [review]
Works fine :)
